modules/core/src/main/scala/org/scalasteward/core/coursier/CoursierAlg.scala

```diff
@@ -0,0 +1,56 @@
+package org.scalasteward.core.coursier
+
+import java.util.concurrent.ExecutorService
+
+import cats.effect._
+import cats.implicits._
+import org.scalasteward.core.data.Dependency
+
+trait CoursierAlg[F[_]] {
+  def getProjectHomepage(dependency: Dependency): F[Option[String]]
+}
+
+object CoursierAlg {
+  def create[F[_]](
+      implicit
+      F: Sync[F]
+  ): CoursierAlg[F] = {
+    implicit val coursierSync: coursier.util.Sync[F] = new coursier.util.Sync[F] {
```

Comment:
This anonymous sub class of `coursier.util.Sync[F]` works but may be not efficient...
I could not figure out a good way to inter-operate with `cats` and `coursier`.

Reply:
With a `cats.effect.ContextShift[F]`, `import coursier.interop.cats._` should automatically provide a `coursier.util.Sync[F]`.

Reply:
Hmm, I tried below 
```scala
    import coursier.interop.cats._
    implicit val cs = cats.effect.ContextShift[F]
    implicit val coursierSync: coursier.util.Sync[F] = implicitly
    val cache = coursier.cache.FileCache[F]()
    val fetch = coursier.Fetch[F](cache)(coursierSync)
```
but
```
[error] ./IdeaProjects/Scala-Steward/modules/core/src/main/scala/org/scalasteward/core/coursier/CoursierAlg.scala:33:47: Cannot find an implicit value for ContextShift[F]:
[error] * import ContextShift[F] from your effects library
[error] * if using IO, use cats.effect.IOApp or build one with cats.effect.IO.contextShift
[error]     implicit val cs = cats.effect.ContextShift[F]
[error]                                               ^
```

Reply:
IIUC, `coursier.interop.cats.catsIOSync` accepts only `ContextShift[IO]`, not `ContextShift[F]`, so I think there is no automatic way right now.

Reply:
Indeed, https://github.com/coursier/coursier/pull/1294 should help.